tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -262,7 +268,12 @@ private void ConfigureTraceTransport(string agentUri, string tracesPipeName, int
             {
                 TracesTransport = TracesTransportType.WindowsNamedPipe;
                 TracesPipeName = tracesPipeName;
-                SetAgentUri(agentHost ?? DefaultAgentHost, agentPort ?? DefaultAgentPort); // this one can throw
+
+                // Still build the Uri no matter what the transport as we send it in the http message
+                // TBH, I don't know if we should handle the case where we use agentHost or agentPort.
+                // Can user have configured both agenthost, agentport, and a UDS path (or an url or a pipe)?
+                // I allow this one to throw, as this was the previous behaviour and because I don't know what to do.
```

Comment:
I don't think we should ever throw here🤔 If they've set up something incorrectly, and it throws, then I think we could stop their app starting, because this will run in `Instrumentation.Initialize()`

Reply:
I'll TryCreate the Uri for retro compatibility then even though we don't need it anymore